src/FTPObject.jl

```diff
@@ -132,168 +132,156 @@ end
-`ftp_dir`: The directory on the ftp server where we want to drop the files.
-`retry_callback::Function=(count, options) -> (count < 4, options)`: Function for retrying
-    when delivery fails.
-`retry_wait_seconds::Integer = 5`: How many seconds to wait in between retries.
+`local_path::AbstractString`: The file path to the file we want to deliver.
+
+# Keywords
+`ftp_options=ftp.ctxt`: FTP Options
+`mode::FTP_MODE=binary_mode`: Set the ftp mode.
+`verbose=nothing`: Set the verbosity
 
 # Returns
-- `Array{Bool,1}`: Returns a vector of booleans with true for each successfully delivered
-                   file and false for any that failed to transfer.
+`Bool`: Returns a boolean with true if the file was successfully transfered, else false.
 """
 function upload(
     ftp::FTP,
-    local_file_paths::Vector{<:AbstractString},
-    ftp_dir::AbstractString;
-    retry_callback::Function=(count, options) -> (count < 4, options),
-    retry_wait_seconds::Integer=5,
-    verbose=nothing,
-)
+    local_path::AbstractString;
+    ftp_options=ftp.ctxt,
+    mode::FTP_MODE=binary_mode,
+    verbose=nothing
+)::Bool
     _dep_verbose_kw(verbose, typeof(ftp), :upload)
-    successful_delivery = Bool[]
-
-    ftp_options = ftp.ctxt
-
-    for single_file in local_file_paths
-        # The location we are going to drop the file in the FTP server
-        server_location = joinpath(ftp_dir, basename(single_file))
-
-        # ftp_put requires an io so open up our file.
-        open(single_file) do single_file_io
-            # Whether or not the current file was successfully delivered to the FTP
-            file_delivery_success = false
-
-            attempts = 1
-            # The loops should break after an appropriate amount of retries.
-            # This way of doing retries makes testing easier.
-            # Defaults to 4 attempts, waiting 5 seconds between each retry.
-            while true
-                try
-                    resp = ftp_put(
-                        ftp_options, server_location, single_file_io; verbose=verbose
-                    )
-                    file_delivery_success = resp.code == complete_transfer_code
-                    if file_delivery_success
-                        break
-                    end
-                catch e
-                    @warn(e)
-                end
-                sleep(retry_wait_seconds)
-                # It returns ftp_options for testing purposes, where the ftp server
-                # starts not existing then comes into existence during retries.
-                do_retry, ftp_options = retry_callback(attempts, ftp_options)
-                if !do_retry
-                    break
-                end
-                attempts += 1
-            end
-
-            push!(successful_delivery, file_delivery_success)
 
-        end
-    end
-
-    return successful_delivery
+    return upload(
+        ftp, local_path, local_path;
```

Comment:
Is this correct? What if `local_path` is `a/b/c`?

Reply:
That's how one of the functions in the original was defined https://github.com/invenia/FTPClient.jl/blob/master/src/FTPObject.jl#L132-L149
I figure the idea is that your local file and remote file ideally have the same structure, which would just end up calling the local_path, remote_path version of upload

Reply:
That being said, perhaps that shouldn't be a function as it could be a bit ambiguous. It might seem like it's just uploading local_file to the cwd of the ftp when in fact it's uploading it to the same path.